modules/ezcontent_preview/src/Controller/PreviewView.php

```diff
@@ -3,37 +3,71 @@
 namespace Drupal\ezcontent_preview\Controller;
 
 use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
 use Drupal\node\NodeInterface;
-use Drupal\Core\Routing\TrustedRedirectResponse; 
-use Drupal\Core\Entity\EntityTypeManager;
-use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Routing\TrustedRedirectResponse;
 use Drupal\Core\Access\AccessResult;
 use Drupal\Core\Session\AccountInterface;
-
 use Drupal\ezcontent_preview\Utils;
-
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
- * Preview the content
+ * Preview the content.
  */
 class PreviewView extends ControllerBase {
 
-  public function preview(NodeInterface $node = NULL, $preview_type) {
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The current route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, RouteMatchInterface $routeMatch) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->routeMatch = $routeMatch;
+  }
 
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('current_route_match')
+    );
+  }
+
+  // Based on the checbox it eturns an iframe output on the.
```

Comment:
Why do we have this comment here ? It is outside the scope of a method / variable.